drivers/scd30/scd30.c

```diff
@@ -0,0 +1,379 @@
+        DEBUG("[scd30] Ambient pressure compensation value out of range\n");
+        return SCD30_INVALID_VALUE;
+    }
+
+    ret = scd30_get_param(dev, SCD30_INTERVAL, &curr_interval);
+    if (ret != 0) {
+        return ret;
+    }
+    if (curr_interval != SCD30_MIN_INTERVAL) {
+        ret = scd30_set_param(dev, SCD30_INTERVAL, SCD30_MIN_INTERVAL);
+        if (ret != 0) {
+            DEBUG("[scd30] Error setting interval to min\n");
+            return ret;
+        }
+    }
+
+    /* Doing this to reduce the traffic on the I2C bus as
+     * compared when constantly polling the device for status
+     */
+    xtimer_sleep(SCD30_MIN_INTERVAL);
+
+    initial_time = xtimer_now_usec();
+    ret = scd30_get_param(dev, SCD30_STATUS, &state);
+    if (ret != 0) {
+        return ret;
+    }
+
+    while (state != 1 &&
+           (xtimer_now_usec() - initial_time) < SCD30_DATA_RDY_TIMEOUT) {
+        ret = scd30_get_param(dev, SCD30_STATUS, &state);
+        if (ret != 0) {
+            return ret;
+        }
+    }
+    ret = _scd30_read_data(dev, result);
+    if (ret != SCD30_OK) {
+        DEBUG("[scd30] Error reading data \n");
+        return ret;
+    }
+
+    if (curr_interval != SCD30_MIN_INTERVAL) {
+        ret = scd30_set_param(dev, SCD30_INTERVAL, curr_interval);
+        if (ret != 0) {
+            DEBUG("[scd30] Error resetting interval to original value\n");
+            return ret;
+        }
+    }
+    return SCD30_OK;
+}
+
+uint8_t scd30_read_periodic(scd30_t *dev, scd30_measurement_t *result)
+{
+    uint16_t state = 0;
+    uint32_t initial_time;
+    int ret = 0;
+
+    initial_time = xtimer_now_usec();
+    ret = scd30_get_param(dev, SCD30_STATUS, &state);
+    if (ret != 0) {
+        return ret;
+    }
+
+    while (state != 1 &&
+           (xtimer_now_usec() - initial_time) < SCD30_DATA_RDY_TIMEOUT) {
+        scd30_get_param(dev, SCD30_STATUS, &state);
+    }
+
+    if(state != 1) {
+        return SCD30_NO_NEW_DATA;
+    }
+    ret = _scd30_read_data(dev, result);
+    if (ret != SCD30_OK) {
+        DEBUG("[scd30] Error reading values");
+        return ret;
+    }
+    return SCD30_OK;
+}
+
+int scd30_start_periodic_measurement(scd30_t *dev, uint16_t *interval,
+                                     uint16_t *apc)
+{
+    int ret = 0;
+
+    /* Check if input is valid */
+    if (!_is_valid_interval(interval)) {
+        DEBUG("[scd30] Interval value out of range\n");
+        return SCD30_INVALID_VALUE;
+    }
+    if (!_is_valid_press_comp(apc)) {
+        DEBUG("[scd30] Ambient pressure compensation value out of range\n");
+        return SCD30_INVALID_VALUE;
+    }
+
+    ret = scd30_set_param(dev, SCD30_INTERVAL, *interval);
+    if (ret != 0) {
+        return ret;
+    }
+    ret = scd30_set_param(dev, SCD30_START, *apc);
+    if (ret != 0) {
+        return ret;
+    }
+    return SCD30_OK;
+}
+
+int8_t scd30_stop_measurements(scd30_t *dev)
+{
+    uint8_t buffer[2];
+    uint16_t cmd = SCD30_STOP;
```

Comment:
```suggestion
    const uint16_t cmd = htons(SCD30_STOP);
```

Now you can 

```C
i2c_write_bytes(SCD30_I2C, SCD30_I2C_ADDRESS, &cmd, sizeof(cmd), 0);
```

Reply:
I understand why you intend to do this. I would have to change this in the `scd30_set_param()` and `scd30_get_param()` functions as well, wouldn't I ?

Reply:
The sensor expects the commands to be sent in big endian format. So, when I do :
```
    buffer[0] = (uint8_t)(cmd >> 8);
    buffer[1] = (uint8_t)cmd;
```
I am taking care of sending the sensor the values in the expected format where the MSB of command is sent first.
For example, in case of SCD30_STOP, i.e. `0x0104`, buffer[0] = 0x01 and buffer[1] - 0x04(which is what the sensor expects too). Therefore, could you help me understand why would I require to use `htons()` when the data format required is already being transmitted with the current code?

Reply:
I was only suggesting that you don't need a separate buffer and don't have to assemble the buffer yourself, you can just use `htons` and then transmit `cmd` directly.

Reply:
Oh yeah, makes sense